downstairs/src/dump.rs

```diff
@@ -236,11 +236,7 @@ fn show_extent(
     }
     print!(" ");
     for (index, _) in region_dir.iter().enumerate() {
-        print!(" {0:^6}", format!("NONCE{}", index));
-    }
-    print!(" ");
-    for (index, _) in region_dir.iter().enumerate() {
-        print!(" {0:^4}", format!("TAG{}", index));
+        print!(" {0:^6}", format!("ECTX{}", index));
```

Comment:
What do you think about an additional flag to the dump command, say `--block <offset>`, that does something like this:

```
$ dump -d disk1/ -d disk2/ -d disk3/ -e 0 -b 7

Block 7:

Data 0 c481f8d9028ab10a22676b1e847e09b710e848686195f4d827e0ad0500f3
Data 1 c481f8d9028ab10a22676b1e847e09b710e848686195f4d827e0ad0500f3
Data 2 a1fe39813ad9bfb55c955c83cad71541df9432f4a0034893aec285773fc7

Nonces               0                         1                         2
     1  e5d61e20365c00d53620c5cf  e5d61e20365c00d53620c5cf  e5d61e20365c00d53620c5cf
     2  680922f9f7c589b1df04494a  680922f9f7c589b1df04494a  680922f9f7c589b1df04494a
     3  7e46ceb38fd91a8ee0fa47ec  7e46ceb38fd91a8ee0fa47ec  7e46ceb38fd91a8ee0fa47ec
     4  38be130a98144e69c42dcc67  38be130a98144e69c42dcc67  38be130a98144e69c42dcc67
     5                                                      be8beb3c40bf226998816f04
                                                             
Tags                    0                                 1                                 2
     1  de978e890af24ca8c3df153c644760cb  de978e890af24ca8c3df153c644760cb  de978e890af24ca8c3df153c644760cb
     2  bfd39ee9fc8ecb84a31ff9b1a3982b42  bfd39ee9fc8ecb84a31ff9b1a3982b42  bfd39ee9fc8ecb84a31ff9b1a3982b42
     3  906a408673b3025b8272cd73e26496aa  906a408673b3025b8272cd73e26496aa  906a408673b3025b8272cd73e26496aa
     4  7efa31fef46f4e4e3d63197388090cd2  7efa31fef46f4e4e3d63197388090cd2  7efa31fef46f4e4e3d63197388090cd2
     5                                                                      bec9e0122acfe4d6d99f58df3c8c947a
```

Reply:
Is this showing, for extent 0 block 7:
 * The three rows are a hash of the data for each `-d` region.
 * The Nonces for this e0:b7, in the example four for disk1 and dis2, and 5 for disk3?
 * The Tags for e0:b7, in the example four for disk1 and dis2, and 5 for disk3?
 
Yeah, that is actually a good level of detail.

Some questions though, is the `-b` always relative to the extent, meaning it would be the 7th block in the 0th extent?  I.e. you could not enter the 756th block directly?

The Nonce and Tag index starts from 1, should we have them start from zero like the block and extent numbers,  unless this is how SQL treats them internally?

If we do display it this way, could we add a column to the extent only output that showed the length of the encryption context for that block?  Just as a clue to know how deep that EC is?

Reply:
> Some questions though, is the -b always relative to the extent, meaning it would be the 7th block in the 0th extent? I.e. you could not enter the 756th block directly?

For now I think this makes sense - I imagine going from `dump -d disk1/ -d disk2/ -d disk3/ -e 0` to seeing that block 7 has a difference to appending `-b 7` to see what that difference is.

> The Nonce and Tag index starts from 1, should we have them start from zero like the block and extent numbers, unless this is how SQL treats them internally?

Yeah I can do that

> If we do display it this way, could we add a column to the extent only output that showed the length of the encryption context for that block? Just as a clue to know how deep that EC is?

:+1:

Reply:
Yeah, that's good, do it!

Reply:
Here's the format as of 112a2ea:
```
     Running `target/debug/crucible-downstairs dump -d disks/d1/ -d disks/d2/ -d disks/d3/ -e 0 -b 1`
Extent 0 Block 1

  DATA  SHA256                                                            VER
     0  b855e293ff1be9de289c61098fd5ea8b2dd6530c901e6d6d6e728b645cef9690   A 
     1  0a503d29e6d6bd782ae14cf118b4567263650fe65b7de9c82f42f74759e09152   B 
     2  b855e293ff1be9de289c61098fd5ea8b2dd6530c901e6d6d6e728b645cef9690   A 

NONCES             0                        1                        2              DIFF 
     0  710e190ac68aec889b9dd9f9 710e190ac68aec889b9dd9f7 710e190ac68aec889b9dd9f9  <--- 
     1                                                    710e190ac68aec889b9dd9f0  <--- 

  TAGS                 0                                1                                2                  DIFF 
     0  193909da477762838716f39274bbf8f3 193909da477762838716f39274bbf8f3 193909da477762838716f39274bbf8f3 
     1                                                                    193909da477762838716f39274bbf8f2  <--- 
```

Reply:
This looks good.
Will the "extent level" output still report a mismatch if a block has nonce/tag mismatches?